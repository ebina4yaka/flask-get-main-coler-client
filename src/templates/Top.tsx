import { ReactElement } from 'react'
import { Box, Container, createStyles } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import Copyright from '../atoms/Copyright'
import Title from '../atoms/Title'
import UploadImage from '../organisms/UploadImage'

const useStyles = makeStyles(
  createStyles({
    component: {
      textAlign: 'center',
    },
    copyright: {
      marginTop: '10px',
    },
  })
)

export default function Top(): ReactElement {
  const classes = useStyles()
  return (
    <Container maxWidth="sm">
      <Box my={4} className={classes.component}>
        <Title />
        <UploadImage />
        <div className={classes.copyright}>
          <Copyright />
        </div>
      </Box>
    </Container>
  )
}
