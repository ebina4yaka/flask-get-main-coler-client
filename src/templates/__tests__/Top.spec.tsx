import renderer from 'react-test-renderer'
import Top from '../Top'

jest.mock('../../atoms/Title', () => 'MockedTitle')
jest.mock('../../organisms/UploadImage', () => 'MockedUploadImage')
jest.mock('../../atoms/Copyright', () => 'MockedCopyright')

test('Snapshot test for Top', () => {
  const tree = renderer.create(<Top />).toJSON()
  expect(tree).toMatchSnapshot()
})
