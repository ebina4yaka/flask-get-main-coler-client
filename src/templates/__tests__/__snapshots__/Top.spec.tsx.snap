// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Top 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthSm"
>
  <div
    className="MuiBox-root MuiBox-root-3 makeStyles-component-1"
  >
    <MockedTitle />
    <MockedUploadImage />
    <div
      className="makeStyles-copyright-2"
    >
      <MockedCopyright />
    </div>
  </div>
</div>
`;
